service/geopmdpy/system_files.py

```diff
@@ -570,6 +572,91 @@ def set_watch_id(self, client_pid, watch_id):
         self._sessions[client_pid]['watch_id'] = watch_id
         self._update_session_file(client_pid)
 
+    def get_reference_count(self, client_pid):
+        """Query the reference count for the client session.
+
+        The reference count is basically just an abstraction for the client.
+        The client itself cannot have more than a single open session at once,
+        since there is a one to one mapping between client pid and open sessions,
+        and lifetime of our session is coupled to the lifetime of the connection.
+
+        The reference count is used when in the client process, we have independent components,
+        which all may request that same connection as a kind of shared resource.
+        We want to provide that same connection as a singleton, and keep track of
+        how many independent components have access to that same connection as a resource,
+        in order to know when to deallocate that resource when all these components
+        have disconnected from the session.
+
+        Args:
+            client_pid (int): Linux PID that opened the session
+
+        Returns:
+            int: GLib watch ID to track the session lifetime
+
+        Raises:
+            RuntimeError: Client does not have an open session
+
+        """
+        self.check_client_active(client_pid, 'get_reference_count')
+        return self._sessions[client_pid]['reference_count']
+
+    def set_reference_count(self, client_pid, reference_count):
```

Comment:
This interface only seems to be used in the test.  We can leave it, but I'm not sure how it would be used in practice.